common/utils/src/main/resources/error/error-classes.json

```diff
@@ -4457,6 +4457,11 @@
           "Expressions referencing the outer query are not supported outside of WHERE/HAVING clauses: <sqlExprs>."
         ]
       },
+      "LAMBDA_OR_HIGHER_ORDER_FUNCTION" : {
```

Comment:
hmmm, are these two the same thing?

Reply:
Reading the code comment
```
/**
 * A higher order function takes one or more (lambda) functions and applies these to some objects.
 * The function produces a number of variables which can be consumed by some lambda function.
 */
```

I think for use-facing error message, we can just say "higher order functions". For the code change, we can check both `HigherOrderFunction` and `LambdaFunction` to  be safe.

Reply:
Sounds good, this is simpler. Done.